### ISSUE TITLE
Fix assertions for secret-handshake

### ISSUE THREAD

**Issue by moniisek** [exercism/java#2060]
The signature of the method is defined as: 
```
List<Signal> calculateHandshake(int number)
```
but we're not given any instructions on how the `Signal` type should look like to pass the tests. My tests keep failing because of incompatible types, for example 
```
expected: java.util.Collections$SingletonList<[JUMP]> but was: java.util.ArrayList<[JUMP]>
```
or 
```
expected: java.util.Arrays$ArrayList<[WINK, DOUBLE_BLINK]> but was: java.util.ArrayList<[WINK, DOUBLE_BLINK]>
```
I'm not sure if this is part of the exercise I should figure out? Thank you for guidance.

**Comment by ericbalawejder** [comment]
Are you running the tests on your local machine or in the Exercism.org code editor?

**Comment by jmrunkle** [comment]
These assertions look bad IMO: https://github.com/exercism/java/blob/a7312ccf3c4dfc88a91bfd6f6dad036695125855/exercises/practice/secret-handshake/src/test/java/HandshakeCalculatorTest.java

These are making assertions that a specific *type* of list is being returned rather than just asserting the contents are correct / have the correct order. Switching to AssertJ better allows us to do the correct thing here with assertions more like this:

```java
assertThat(handshakeCalculator.calculateHandshake(NNN))
    .containsExactly(...);
```

So, in short, we should fix the assertions.

**Comment by moniisek** [comment]
Thank you!
